code/controllers/subsystem/fail2topic.dm

```diff
@@ -88,10 +88,10 @@ SUBSYSTEM_DEF(fail2topic)
 /datum/controller/subsystem/fail2topic/proc/BanFromFirewall(ip)
 	if (!enabled)
 		return
-	var/static/regex/R = regex(@"(\.0\.)|(\.0$)|(\l+|/|;|&|\||-|%)") // Anything that interacts with a shell should be parsed. Prevents direct call input tampering
-	if(length(ip) > 15 || length(findtext(ip, R)))
+	var/static/regex/R = regex(@"^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$") // Anything that interacts with a shell should be parsed. Prevents direct call input tampering
+	ip = findtext(ip, R, "")
```

Comment:
looking at the ref, isn't the third argument of findtext supposed to be a starting index? Also I think 0.0.0.0 and 127.0.0.1 are valid matches for this expression

Reply:
0.0.0.0 won't block all IPs in netsh without a subnet mask, 0.0.0.0/0 will though and regex would catch that

![image](https://user-images.githubusercontent.com/59686430/133566989-8e6ae7b5-e244-47b0-a2c3-02dbed777b60.png)

Reply:
default CIDR is /32